src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -145,6 +149,20 @@ private ExecutableElement findFirst(Collection<ExecutableElement> elements, Stri
         + "}\n");
   }
 
+  @Test public void overrideDoesNotCopyDefaultModifier() {
+    TypeElement classElement = getElement(ExtendsIterableWithDefaultMethods.class);
+    DeclaredType classType = (DeclaredType) classElement.asType();
+    List<ExecutableElement> methods = methodsIn(elements.getAllMembers(classElement));
+    ExecutableElement exec = findFirst(methods, "iterator");
+    Assume.assumeTrue(Util.DEFAULT != null);
```

Comment:
`assumeNotNull`?

Also maybe `import static`? (and possibly use `TruthJUnit.assume()` instead: `assume().that(Util.DEFAULT).isNotNull()`)